### PR TITLE
Add Strings.equal

### PR DESCRIPTION
**Pull request by 0xCaso**
Hi! I'd like to propose the addition of the new method ```equal``` for Strings, which would make things more elegant and make the code less repetitive. Also, it should be really useful for testing purposes with Foundry.

I'm referring to this (old) issue: https://github.com/OpenZeppelin/openzeppelin-contracts/issues/1795#issuecomment-1286701350

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Changelog entry

### CODE REVIEW

File: contracts/utils/Strings.sol [hunk 1]
```diff
@@ -67,4 +67,12 @@ library Strings {
     function toHexString(address addr) internal pure returns (string memory) {
         return toHexString(uint256(uint160(addr)), _ADDRESS_LENGTH);
     }
+
+    /**
+     * @dev Returns true if the two strings are equal.
+     */
+    function compare(string memory a, string memory b) internal pure returns (bool) {
+        if (bytes(a).length != bytes(b).length) return false;
+        return keccak256(abi.encodePacked(a)) == keccak256(abi.encodePacked(b));
```
Comment (Amxx) [inline contracts/utils/Strings.sol:76]: Lets make use of lazy evaluation (also, the packing is not needed, and expensive)
```suggestion
        return bytes(a).length == bytes(b).length && keccak256(a) == keccak256(b);
```

Comment (Amxx) [reply contracts/utils/Strings.sol:76]: I'm curious how much the length check cost compared to what it saves. Is it really worth it ?

Comment (0xCaso) [reply contracts/utils/Strings.sol:76]: Actually, the compiler gives error if I try with `keccak256(a) == keccak256(b)`, am I doing something wrong? Also, [here](https://github.com/0xCaso/test-strings-compare-function) I compared four different functions, and the below one seems the winner.
```suggestion
        return keccak256(bytes(a)) == keccak256(bytes(b));
```

**Comment by Amxx** [comment]
Note: I'd love #3666 to be merge first, so that the tests can use the new syntax (instead of having to update #3666 yet again)

**Comment by 0xCaso** [comment]
What's the state of #3666? @Amxx

File: contracts/utils/Strings.sol [hunk 2]
```diff
@@ -67,4 +67,11 @@ library Strings {
     function toHexString(address addr) internal pure returns (string memory) {
         return toHexString(uint256(uint160(addr)), _ADDRESS_LENGTH);
     }
+
+    /**
+     * @dev Returns true if the two strings are equal.
+     */
+    function compare(string memory a, string memory b) internal pure returns (bool) {
```
Comment (frangio) [inline contracts/utils/Strings.sol:74]: Can we rename this to `equal`?

Comment (0xCaso) [reply contracts/utils/Strings.sol:74]: Sure!

**Review by frangio — DISMISSED** [review]
LGTM

**Review by frangio — APPROVED** [review]

**Review by ernestognw — APPROVED** [review]
LGTM.

I'd like to explorer further if there's a cheaper way of doing this since keccak is an expensive operation.

I don't think it's worth it at the moment but mentioning for the record.

**Pull request merged**